articles/internet-inconsistencies-with-r-programming/index.md

```diff
@@ -0,0 +1,288 @@
+  CIRA | Canadian Internet Registration Authority serves Canada.
+
+- Privacy redactions
+
+The country classified with this IP address is complex. Hostinger International Limited (AS47583) is the ASN hosting website responsible for IP addresses between `45.88.197.0` to `45.88.197.255`.
+
+With reverse IP engineering done on `45.88.197.212`, we found five possible geological locations:
+- Lithuania (Li)
+- Cyprus (Cyp)
+- Germany (De)
+- Netherlands (Nl)
+- Amsterdam (Am)  
+
+`Rattle` can generate data models. A decision tree model can provide a logical breakdown. Shown below, is a manually made IP address data frame:
+![dataframe](/engineering-education/internet-inconsistencies-with-r-programming/mainlocationsforasn47583.jpg)  
+
+Typically, a decision tree selects the highest possible number as the optimal choice.
+
+In this scenario, the countries categorized as less optimal are analyzed. `Amsterdam`, `Netherlands`, and `Cyprus` were shown as the top three choices. `Lithuania` and `Germany` seemed to be less optimal.
+![decisiontree](/engineering-education/internet-inconsistencies-with-r-programming/rplot.jpeg)  
+*Screen capture*  
+
+It is possible to evaluate variable importance from a random forest model. Variable importance is shown in the image below:
+![variableimportance](/engineering-education/internet-inconsistencies-with-r-programming/variableimportance.jpg)  
+*Screen capture*  
+
+With the highest score of the five countries, `Lithuania` showed the most links to the IP address. `Germany` also showed some correlation. This statistical analysis using Gini found `Lithuania` generated higher variable importance with a value of `3087.48`.
+
+### Linux reverse IP lookup
+To verify validity, here is a quick code to assess:  
+
+```bash
+sudo curl http://ipinfo.io/45.88.197.212
+```  
+
+**Output:**
+
+```bash
+{
+  "ip": "45.88.197.212",
+  "city": "Kaunas",
+  "region": "Kaunas",
+  "country": "LT",
+  "loc": "54.9027,23.9096",
+  "org": "AS47583 Hostinger International Limited",
+  "postal": "44001",
+  "timezone": "Europe/Vilnius",
+  "readme": "https://ipinfo.io/missingauth"
+}
+```  
+
+![ipinfolinux](/engineering-education/internet-inconsistencies-with-r-programming/ipinfolinux.jpg)  
+*Screen capture*
+
+A `curl` function can list the possible domain names on an IP address. The code below uses reverse IP engineering.
+
+```bash
+sudo curl https://host.io/asianausa.us
+```  
+
+**Partial Output:**
+
+```bash
+grh-interviews.online
+recruits-agility.com
+careers-mfc.work
+careers-massiveinsights.work
+grandrivershospital.com
+mindfieldconsulting.work
+careers-mconsulting.work
+grandrivershosp.ca
+interviews-sobeys.com
+interviews-massiveinsights.digital
+morgeesmodcon.com
+```
+
+Did you notice the domain names listed above are companies registered with `ARIN` and `CIRA` without any connection to `RIPE`?  
+
+Internet inconsistencies exist.
+
+### Takeaways
```

Comment:
Though you have added the takeaways, it would be great if you give a proper conclusion of what was discussed previously, before specifying the takeaways.